### PR TITLE
Updates the documentation of the ownership.

### LINKED ISSUE #281: Document ownership model
inside of README.md

### PR DESCRIPTION
**Pull request by SteffenSeckler**
# Description

Enhances the documentation of the ownership model used by AutoPas.

## Related Pull Requests

- #278 

## Resolved Issues

- [x] fixes #281 

# <strike>How Has This Been Tested?</strike>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -181,12 +178,24 @@ Analogously to `begin()`, `cbegin()` is also defined, which guarantees to return
...
+   This call will potentially trigger an update of the container inside of AutoPas. The update will be performed if either
+   
+    a. Enough samples are collected for the current configuration OR
+    
+    b. The rebuild frequeny is reached.
+   
+   If the update is performed, the returned bool `updated` is `true` and the returned vector `invalidParticles` consists of the particles that are leaving this container. These are particles which were previously owned by this AutoPas container, but have left the boundary of this container, i.e., their current position resides outside of the container.
+   
+   If the update is not performed, `updated` will be false and the returned vector `invalidParticles` will be empty.
+   An update is sometimes skipped to ensure that containers do not change and to enable the best possible performance for containers that use neighbor lists.
```
Comment (FG-TUM) [inline README.md:194]: Justify why we want containers sometimes not to change.

File: README.md [hunk 2]
```diff
@@ -181,12 +178,24 @@ Analogously to `begin()`, `cbegin()` is also defined, which guarantees to return
...
 
 2. Handling the leaving particles
+   * This step can be skipped if `updated` was false. If you use multiple MPI instances, you have to ensure that all instances rebuild at the same time steps. This is guaranteed, if the sampling frequency is the same as (or a multiple of) the rebuild frequency.
```
Comment (FG-TUM) [inline README.md:197]:
```suggestion
   * This step can be skipped if `updated` was false. If you use multiple MPI instances, you have to ensure that all instances rebuild during the same time step. This is guaranteed if the sampling frequency is the same as (or a multiple of) the rebuild frequency.
```

File: README.md [hunk 3]
```diff
@@ -181,12 +178,24 @@ Analogously to `begin()`, `cbegin()` is also defined, which guarantees to return
...
+   This call will potentially trigger an update of the container inside of AutoPas. The update will be performed if either
+   
+    a. The AutoTuner collected enough samples for the current configuration and will move to the next one OR
+    
+    b. The rebuild frequency of the container is reached.
+   
+   If the update is performed, the returned bool `updated` is `true`. The returned vector `invalidParticles` consists of the particles that are not anymore within the boundaries of this container and hence are deleted from it. These are particles that were previously owned by this AutoPas container but have left the boundary of this container, i.e., their current position resides outside of the container.
+   
+   If the update is not performed, `updated` will be false and the returned vector `invalidParticles` will be empty.
+   An update is sometimes skipped to ensure that containers do not change and to enable the best possible performance for containers that use neighbor lists.
```
Comment (SteffenSeckler) [inline README.md:194]:
```suggestion
   If the update is not performed, `updated` will be false and the returned vector `invalidParticles` will be empty.
   An update is sometimes skipped to ensure that containers do not change, which allows containers to reuse neighbor lists thus enabling better performance.
```
@FG-TUM is this fine by you?

Comment (FG-TUM) [reply README.md:194]: yes

File: README.md [hunk 4]
```diff
@@ -119,9 +119,9 @@ TODO
...
+Particles saved in an AutoPas container can be one of two possible states:
+* owned: Particles that belong to this AutoPas instance. These particles are either inside of the boundary of the AutoPas instance or very close to the boundary (less than a distance of skin/2 away). If a particle is added via `addParticle()`, it is automatically added as an owned particle. An owned particle can explicitly be removed by deleting the particle using an iterator (`autoPas.deleteParticle(iterator)`). On update of the AutoPas container (using `updateContainer()`) owned particles that move outside of the boundary of its parent, AutoPas container are returned.
```
Comment (FG-TUM) [inline README.md:123]:
```suggestion
* owned: Particles that belong to this AutoPas instance. These particles are either inside of the boundary of the AutoPas instance or very close to the boundary (less than a distance of skin/2 away). If a particle is added via `addParticle()`, it is automatically added as an owned particle. An owned particle can explicitly be removed by deleting the particle using an iterator (`autoPas.deleteParticle(iterator)`). On update of the AutoPas container (using `updateContainer()`) owned particles that move outside of the boundary of its parent AutoPas container are returned.
```
(the `,` between parent and AutoPas at the end is misleading imho)

**Pull request merged**